src/lib/operators/get_table.cpp

```diff
@@ -23,5 +28,27 @@ std::shared_ptr<AbstractOperator> GetTable::recreate(const std::vector<AllParame
   return std::make_shared<GetTable>(_name);
 }
 
-std::shared_ptr<const Table> GetTable::_on_execute() { return StorageManager::get().get_table(_name); }
+std::shared_ptr<const Table> GetTable::_on_execute() {
+  auto original_table = StorageManager::get().get_table(_name);
+  if (!_excluded_chunks || _excluded_chunks->empty()) {
+    return original_table;
+  }
+  auto pruned_table = Table::create_with_layout_from(original_table, original_table->max_chunk_size());
```

Comment:
I think a quick comment stating that the original table is copied, excluding the chunks that can be pruned, could help get a better overview

Reply:
done